In [50]:
# lets import some required library

import pandas as pd
import numpy as np

In [51]:
# lets import the movies data 

movies = pd.read_csv('./movielens/movies.csv' , sep=',')
print(type(movies))
movies.head()

<class 'pandas.core.frame.DataFrame'>


movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [53]:
#We can see that this movies.csv file has three variables "movieId", "title", and "genres". Next lets check the number of rows it has 
movies.shape

(62423, 3)

In [54]:
# lets import the tags data
tags = pd.read_csv('./movielens/tags.csv' , sep=',')
print(type(tags))
tags.head()

<class 'pandas.core.frame.DataFrame'>


userId  movieId               tag   timestamp
0       3      260           classic  1439472355
1       3      260            sci-fi  1439472256
2       4     1732       dark comedy  1573943598
3       4     1732    great dialogue  1573943604
4       4     7569  so bad it's good  1573943455

In [55]:
tags.shape

(1093360, 4)

SO the tag data has four column of variable and has 1093360 rows.

In [ ]:
# now lets import rating file

In [56]:
ratings = pd.read_csv('./movielens/ratings.csv' , sep=',')
print(type(ratings))
ratings.head()

<class 'pandas.core.frame.DataFrame'>


userId  movieId  rating   timestamp
0       1      296     5.0  1147880044
1       1      306     3.5  1147868817
2       1      307     5.0  1147868828
3       1      665     5.0  1147878820
4       1      899     3.5  1147868510

In [57]:
ratings.shape

(25000095, 4)

So ratings has 4 columns and 25000095 samples/rows

In [58]:
links = pd.read_csv('./movielens/links.csv' , sep=',')
print(type(links))
links.head()

<class 'pandas.core.frame.DataFrame'>


movieId  imdbId   tmdbId
0        1  114709    862.0
1        2  113497   8844.0
2        3  113228  15602.0
3        4  114885  31357.0
4        5  113041  11862.0

In [59]:
links.shape

(62423, 3)

So link has 3 columns and 62423 samples/rows

In [5]:
tag1 = tags['tag'].unique().tolist()
len(tag1)

73051

In [7]:
tag1

['classic',
 'sci-fi',
 'dark comedy',
 'great dialogue',
 "so bad it's good",
 'unreliable narrators',
 'tense',
 'artificial intelligence',
 'philosophical',
 'cliche',
 'musical',
 'horror',
 'unpredictable',
 'Oscar (Best Supporting Actress)',
 'adventure',
 'anime',
 'ecology',
 'fantasy',
 'Hayao Miyazaki',
 'Miyazaki',
 'post-apocalyptic',
 'bah',
 'Clint Eastwood',
 'music',
 'art',
 'contemporary art',
 'documentary',
 'aliens',
 'amazing photography',
 'Director: James Cameron',
 'first contact',
 'James Cameron',
 'Michael Biehn',
 'android(s)/cyborg(s)',
 'apocalypse',
 'Arnold Schwarzenegger',
 'franchise',
 'terminator',
 'time travel',
 'science fiction',
 'inferior sequel',
 'setting:London (UK) (future)',
 'unoriginal',
 'bad science',
 'good science',
 'Hans Zimmer',
 'philosophical issues',
 'space',
 'space travel',
 'time-travel',
 'visually appealing',
 'bittersweet',
 'boring',
 'conversation',
 'dialogue driven',
 'loneliness',
 'love story',
 'philosophy',
 'qu

In [10]:
print(tag1.value_counts())

AttributeError: 'list' object has no attribute 'value_counts'

In [6]:
tags.isnull().any()

userId       False
movieId      False
tag           True
timestamp    False
dtype: bool

In [7]:
del tags['timestamp']

In [8]:
tags

userId  movieId                  tag
0             3      260              classic
1             3      260               sci-fi
2             4     1732          dark comedy
3             4     1732       great dialogue
4             4     7569     so bad it's good
...         ...      ...                  ...
1093355  162521    66934  Neil Patrick Harris
1093356  162521   103341     cornetto trilogy
1093357  162534   189169               comedy
1093358  162534   189169             disabled
1093359  162534   189169              robbery

[1093360 rows x 3 columns]

In [23]:
x=tags['tag'].value_counts()
print(x)

tag
sci-fi              8330
atmospheric         6516
action              5907
comedy              5702
surreal             5326
                    ... 
teen sleuth            1
evil twins             1
paternity test         1
QVC                    1
cornetto triolgy       1
Name: count, Length: 73050, dtype: int64


In [37]:
maxValues = x.max(axis=0)
maxValues

8330

In [9]:
movies = pd.read_csv('./movielens/movies.csv' , sep=',')
print(type(movies))
movies.head()

<class 'pandas.core.frame.DataFrame'>


movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [10]:
tags.shape

(1093360, 3)

In [25]:
tags['tag']

0                      classic
1                       sci-fi
2                  dark comedy
3               great dialogue
4             so bad it's good
                  ...         
1093355    Neil Patrick Harris
1093356       cornetto trilogy
1093357                 comedy
1093358               disabled
1093359                robbery
Name: tag, Length: 1093344, dtype: object

In [11]:
tags = tags.dropna()

In [12]:
tags.shape

(1093344, 3)

In [13]:
tagscopy = tags.head(500000)

In [14]:
tagscopy.shape

(500000, 3)

In [15]:
movies.shape

(62423, 3)

In [26]:
tag_counts = tags['tag'].value_counts()
tag_counts.shape

(73050,)

In [17]:
tag_counts.max()

8330

In [18]:
ratings = pd.read_csv('./movielens/ratings.csv' , sep=',')
print(type(ratings))
ratings.head()

<class 'pandas.core.frame.DataFrame'>


userId  movieId  rating   timestamp
0       1      296     5.0  1147880044
1       1      306     3.5  1147868817
2       1      307     5.0  1147868828
3       1      665     5.0  1147878820
4       1      899     3.5  1147868510

In [19]:
del ratings['timestamp']

In [20]:
ratings.head()

userId  movieId  rating
0       1      296     5.0
1       1      306     3.5
2       1      307     5.0
3       1      665     5.0
4       1      899     3.5

In [21]:
ratingcopy = ratings.head(1000000) # copy first 600000 rows into ratingcopy from rating


In [22]:
ratingcopy.columns

Index(['userId', 'movieId', 'rating'], dtype='object')

In [23]:
newdf4 = pd.merge(tagscopy, ratingcopy, on="movieId", how="inner")
newdf4.head(10)

userId_x  movieId      tag  userId_y  rating
0         3      260  classic         2     5.0
1         3      260  classic         3     4.0
2         3      260  classic         4     3.5
3         3      260  classic         5     5.0
4         3      260  classic         6     5.0
5         3      260  classic         8     3.0
6         3      260  classic         9     4.0
7         3      260  classic        10     5.0
8         3      260  classic        12     5.0
9         3      260  classic        13     4.5

In [24]:
newdf4['tag']=newdf4['tag'].astype('category').cat.codes
newdf4['rating']=newdf4['rating'].astype('category').cat.codes
newdf4.corr()

userId_x   movieId       tag  userId_y    rating
userId_x  1.000000  0.007926 -0.052679 -0.000645  0.017467
movieId   0.007926  1.000000 -0.009230  0.009016 -0.041709
tag      -0.052679 -0.009230  1.000000 -0.000629  0.016533
userId_y -0.000645  0.009016 -0.000629  1.000000 -0.006990
rating    0.017467 -0.041709  0.016533 -0.006990  1.000000

In [18]:
ratings.isnull().any()

userId     False
movieId    False
rating     False
dtype: bool

In [19]:
ratings.shape

(25000095, 3)

In [20]:
newdf = pd.merge(movies, ratings, on="movieId", how="inner")
newdf.head(10)

movieId             title                                       genres  \
0        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
1        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
2        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
3        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
4        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
5        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
6        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
7        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
8        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   
9        1  Toy Story (1995)  Adventure|Animation|Children|Comedy|Fantasy   

   userId  rating  
0       2     3.5  
1       3     4.0  
2       4     3.0  
3       5     4.0  
4       8     4.0  
5      10     3.5  
6      12     4.0  
7      13     4.0  
8      18     3.0  
9      26     3.0

In [21]:
#corr = newdf['genres'].corr(newdf['rating'])

In [22]:
#tagsrating = tagsrating.dropna()

In [23]:
#tagsrating

In [25]:
df2

movieId                           title  \
0               1                Toy Story (1995)   
1               1                Toy Story (1995)   
2               1                Toy Story (1995)   
3               1                Toy Story (1995)   
4               1                Toy Story (1995)   
...           ...                             ...   
25000090   209157                       We (2018)   
25000091   209159       Window of the Soul (2001)   
25000092   209163                Bad Poems (2018)   
25000093   209169             A Girl Thing (2001)   
25000094   209171  Women of Devil's Island (1962)   

                                               genres  userId  rating  
0         Adventure|Animation|Children|Comedy|Fantasy       2     3.5  
1         Adventure|Animation|Children|Comedy|Fantasy       3     4.0  
2         Adventure|Animation|Children|Comedy|Fantasy       4     3.0  
3         Adventure|Animation|Children|Comedy|Fantasy       5     4.0  
4         Adventure|Animation|Children|Comedy|Fantasy       8     4.0  
...                                               ...     ...     ...  
25000090                                        Drama  119571     1.5  
25000091                                  Documentary  115835     3.0  
25000092                                 Comedy|Drama    6964     4.5  
25000093                           (no genres listed)  119571     3.0  
25000094                       Action|Adventure|Drama  119571     3.0  

[25000095 rows x 5 columns]

In [26]:
del df2['title']

In [27]:
df2

movieId                                       genres  userId  rating
0               1  Adventure|Animation|Children|Comedy|Fantasy       2     3.5
1               1  Adventure|Animation|Children|Comedy|Fantasy       3     4.0
2               1  Adventure|Animation|Children|Comedy|Fantasy       4     3.0
3               1  Adventure|Animation|Children|Comedy|Fantasy       5     4.0
4               1  Adventure|Animation|Children|Comedy|Fantasy       8     4.0
...           ...                                          ...     ...     ...
25000090   209157                                        Drama  119571     1.5
25000091   209159                                  Documentary  115835     3.0
25000092   209163                                 Comedy|Drama    6964     4.5
25000093   209169                           (no genres listed)  119571     3.0
25000094   209171                       Action|Adventure|Drama  119571     3.0

[25000095 rows x 4 columns]

In [32]:
df2['genres']=df2['genres'].astype('category').cat.codes
df2['rating']=df2['rating'].astype('category').cat.codes
df2.corr()

movieId    genres    userId    rating
movieId  1.000000 -0.052287 -0.004413 -0.009216
genres  -0.052287  1.000000 -0.000726  0.048385
userId  -0.004413 -0.000726  1.000000  0.001940
rating  -0.009216  0.048385  0.001940  1.000000